edb/server/mng_port/edgecon.pyx

```diff
@@ -773,14 +797,35 @@ cdef class EdgeConnection:
             num_fields -= 1
         return attrs
 
+    cdef write_headers(self, buf: WriteBuffer, headers: dict):
+        buf.write_int16(len(headers))
+        for k, v in headers.items():
+            buf.write_int16(<int16_t><uint16_t>k)
+            buf.write_len_prefixed_utf8(str(v))
+
     async def parse(self):
         cdef:
             bint json_mode
             bytes eql
+            dict headers
+            int implicit_limit = 0
```

Comment:
Are you aware that this is C 'int' type (16 bit)? I think you're better off not declaring `implicit_limit` in a `cdef` (that will make it a Python object, and thus it will be of unlimited precision)

Reply:
`int` is 32 bits on most platforms.  I can declare it as `uint64_t`.  Unlimited precision for limit makes little sense, as you are unlikely to get ten quintillion entries in your result :smiley:, and it's better to have defined limits for things in the protocol, even if they're internal/undocumented bits.

Reply:
> I can declare it as uint64_t

Yeas, that would work.